src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientImpl.java

```diff
@@ -91,16 +91,46 @@
     private Gson gson;
     private Map<String, Cache<CacheKey, GraphqlResponse<?, ?>>> caches;
     private GraphqlClientMetrics metrics;
-    private GraphqlClientConfiguration configuration;
+    private GraphqlClientConfigurationImpl configuration;
 
     @Activate
     public void activate(GraphqlClientConfiguration configuration) throws Exception {
-        this.configuration = configuration;
-        gson = new Gson();
-        metrics = metricsRegistry != null
+        this.configuration = new GraphqlClientConfigurationImpl(configuration);
+        this.gson = new Gson();
+        this.metrics = metricsRegistry != null
             ? new GraphqlClientMetricsImpl(metricsRegistry, configuration)
             : GraphqlClientMetrics.NOOP;
 
+        if (this.configuration.socketTimeout() <= 0) {
+            LOGGER.warn("Socket timeout set to infinity. This may cause Thread starvation and should be urgently reviewed. Falling back to "
+                + "default configuration.");
+            this.configuration.setSocketTimeout(GraphqlClientConfiguration.DEFAULT_SOCKET_TIMEOUT);
```

Comment:
The config description of this setting says _"Is it 0 or smaller the configured service will not start."_. Here you actually fallback to the default value and the service would start. Please keep description and behaviour consistent.

Reply:
Good catch, fixed.